### ISSUE TITLE
svelte:head not removing/updating components

### ISSUE THREAD

**Issue by Nerglej** [sveltejs/kit#12365]
#### TL;DR
Seperate <title> and <meta> tags in <svelte:head> is fine, a custom component containing <title> and <meta> tags in <svelte:head> is not fine.

#### The problem
Before we simply placed the title in a <title> tag and metadata in <meta> tags in svelte:head. It worked without flaws, but we were limited.
Now I've created a simple SEO component that takes a SEO object (simplified in REPL below). This SEO component is meant to be placed in svelte:head components on all pages where SEO is provided from our CMS. Before we just used a few datapoints from the page itself like the h1 tag as a <title>, but now we can customize a lot of stuff. Therefor a generalized component were needed.

Now, the SEO itself works fine for a page visit, until we navigate. The previous <title> and <meta> tags aren't removed and the new ones are just appended behind everything and doesn't take effect in the browser level.

I'm not sure if this is intended behavior. If it is, it should be clearly stated in the documentation for either SvelteKit or Svelte. Or even worse, if we're blocked by how the internals work, we should give a clear, and preferably simple, alternative. I'm currently considering a context with SEO info in the root layout, but I haven't experimented with that yet😅

Otherwise, thanks for the amazing framework, I've really enjoyed working with it. This is just a minor problem, I've experienced across a couple of different projects now, so I decided to make an issue.

Have a nice day😄

### Reproduction

`lib/Seo.svelte`
```svelte
<script>
	export let title;
	export let metaDescription;
</script>

<title>{title}</title>

<meta name="description" content={metaDescription} />
<meta property="og:description" content={metaDescription} />
```
`routes/+page.svelte`
```svelte
<script>
	import Seo from '$lib/Seo.svelte'
</script>

<svelte:head>
	<Seo title="Main page" metaDescription="Description for main page" />
</svelte:head>

<h1>Main page</h1>
<a href="/sub">Go to subpage</a>
```

`routes/sub/+page.svelte`
```svelte
<script>
	import Seo from '$lib/Seo.svelte';
</script>

<svelte:head>
	<Seo title="Subpage title" metaDescription="This is a description for the subpage"></Seo>
</svelte:head>

<h1>Subpage</h1>

<a href="/">Go to main page</a>
```

Reproducible repo: https://github.com/Nerglej/sveltekit_head_bug

I also create a SvelteLab REPL if that's okay:
https://www.sveltelab.dev/87889fasa3uab9m

To view the problem in the REPL, use your browser debugger (usually F12), and inspect the iframe where the content is rendered. Under the <html> tag you can find the <head> where each navigation just appends new titles and metatags

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @sveltejs/adapter-auto: ^3.0.0 => 3.2.2 
    @sveltejs/kit: ^2.0.0 => 2.5.16 
    @sveltejs/vite-plugin-svelte: ^3.0.0 => 3.1.1 
    svelte: ^4.0.5 => 4.2.18 
    vite: ^5.0.0 => 5.3.1
```

### Severity

serious, but I can work around it

### Additional Information

I ran the System Info command in the SvelteKit REPL mentioned above, so you should roughly get the same results. I have the same problem on three machines, both Windows and Linux, so I don't suspect the system is the problem but either SvelteKit or Svelte.

**Comment by exorman** [comment]
hi @Nerglej! Did you manage this problem somehow? I've just realized I have same bug. Maybe setting <svelte:head> tags only in top level compoment will fix it?

**Comment by Nerglej** [comment]
I "fixed" it in the sense, that instead of having the seo-data in a component that I put into the svelte:head component, I instead just put the svelte:head inside my component.

So instead of

```html
<svelte:head>
  <Seo ... />
</svelte:head>
```

I do

```html
<Seo> /* as in, the SEO-component-file*/ 
  <svelte:head>
    ...
  </svelte:head>
</Seo>
```

So basically I swapped the the tags.
I'm not too keen with this way of doing it, because I wanted to use my seo component elsewhere, but I put the "NO DUPLICATE CODE" idea behind and saved myself tons of work by just having to almost identical components.

Hope it helped😄

**Comment by RyanWolfen7** [comment]
i'm using on mount and the only one that stays is google recaptcha


<script lang="ts">
	import { onMount, onDestroy } from "svelte";
	let { 
		id,
		type = "button" as "submit" | "button" | "reset" | undefined,
		classNames = "bg-amber-500 hover:bg-amber-600 text-slate-900 font-medium py-2 px-6 rounded-md transition-all flex items-center",
		onClick = () => null,
		children
	} = $props();
	const RECAPTCHA_KEY = '6Ld8gPEqAAAAAHdlGkio9KurLbA1pHF2GM5k66ZJ'
	const updatedClass = `g-recaptcha ${classNames}`
	const onSubmit = `
      function onSubmit(token) {
        document.getElementById('form-to-book-ryan').submit();
      }
    `;
	let grecaptchaScript: HTMLScriptElement;
	let onSubmitScript: HTMLScriptElement;
		

	onMount(() => {
		grecaptchaScript = document.createElement('script');
		grecaptchaScript.src = 'https://www.google.com/recaptcha/api.js';
		grecaptchaScript.async = true;
		grecaptchaScript.defer = true;
		document.head.appendChild(grecaptchaScript);

		onSubmitScript = document.createElement('script');
		onSubmitScript.type = 'text/javascript';
		onSubmitScript.innerHTML = onSubmit;
		document.body.appendChild(onSubmitScript);
	});
	onDestroy(() => {
		if (grecaptchaScript && grecaptchaScript.parentNode) {
			grecaptchaScript.parentNode.removeChild(grecaptchaScript);
		}
		if (onSubmitScript && onSubmitScript.parentNode) {
			onSubmitScript.parentNode.removeChild(onSubmitScript);
		}
	});
</script>


<button
	id={id}
	type={type}
	class={updatedClass}
	data-sitekey={RECAPTCHA_KEY}
	data-callback=onSubmit
	data-action="submit"
>
	{@render children()}
</button>
